stdlib/ripper/0/ripper.rbs

```diff
@@ -458,382 +458,382 @@ class Ripper
     attr_reader error: untyped
 
     alias compile_error on_error
-    def on_BEGIN: (*untyped args) -> untyped
+    def on_BEGIN: (*untyped args) -> Array[untyped]
 
-    def on_CHAR: (untyped tok) -> untyped
+    def on_CHAR: (String tok) -> [Symbol, String, [Integer?, Integer?]]
```

Comment:
These two patterns of callbacks are defined here:

https://github.com/ruby/ruby/blob/master/ext/ripper/lib/ripper/sexp.rb#L116-L131